packages/wonder-blocks-data/util/response-cache.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+import type {
+    CacheEntry,
+    ResponseCache as Cache,
+    IRequestHandler,
+} from "./types.js";
+
+function deepClone<T: {...}>(source: T | $ReadOnly<T>): $ReadOnly<T> {
+    /**
+     * We want to deep clone the source cache to dodge mutations by external
+     * references. So we serialize the source cache to JSON and parse it
+     * back into a new object.
+     */
+    const serializedInitCache = JSON.stringify(source);
+    const cloneInitCache = JSON.parse(serializedInitCache);
+    return Object.freeze(cloneInitCache);
+}
+
+/**
+ * The default instance is stored here.
+ * It's created below in the Default() static property.
+ */
+let _default: ResponseCache;
+
+/**
+ * Implements the response cache.
+ *
+ * INTERNAL USE ONLY
+ */
+export class ResponseCache {
+    static get Default() {
```

Comment:
nit: I wonder if it's worth to rename this method to `static getInstance()` or `static get instance()`. I know this won't affect the implementation at all and its consumers, b/c it's an internal class. So, please disregard this comment if it doesn't add any value.

Reply:
I wanted this to be clear that it is returning the same instance each time, not a new one. I felt `Default` conveyed that better than `Instance`.